ADEN-5041 Fix PF so it doesn't fire when a user does not have an ad blocking software

dianafa opened the pull request:
https://wikia-inc.atlassian.net/browse/ADEN-5041

dianafa committed:
ADEN-5041

reorganize /ARecoveryEngine/js/ dir

dianafa committed:
ADEN-5041

WIP

dianafa committed:
ADEN-5041

move track and verifyContent to adBlockDetection

dianafa committed:
ADEN-5041 use log.levels

dianafa committed:
ADEN-5041

add adonis-marker in gpt helper instead of googleTag

dianafa committed:
ADEN-5041

add marker in helper basing on data from gpt factory

dianafa committed:
ADEN-5041

recover only if recovery enabled

dianafa committed:
ADEN-5041

cleanup log

dianafa committed:
ADEN-5041

fix tests

dianafa committed:
ADEN-5041

add adBlockDetection tests

dianafa committed:
ADEN-5041

remove unused imports

fraszczakszymon commented on extensions/wikia/ARecoveryEngine/js/adBlockDetection.js:
```diff
@@ -1,38 +1,32 @@
 /*global define*/
-define('ext.wikia.aRecoveryEngine.recovery.sourcePoint', [
+define('ext.wikia.aRecoveryEngine.adBlockDetection', [
```
Just wonder if it's safe to use this name :>

fraszczakszymon replied:
IMO this module has too many responsibilities. I'd move `isEnabled`=`isRecoveryEnabled`, `getSafeUri`, `verifyContent`, `track` to another, second abstraction - "recovery". Then AdEngineRunner can really depnend on recovery (which depends on detection) instead of detection directly. Let's keep detection abstraction away from user facing changes

fraszczakszymon commented on extensions/wikia/ARecoveryEngine/js/SourcePoint/sourcePoint.js:
```diff
@@ -0,0 +1,52 @@
+/*global define*/
+define('ext.wikia.aRecoveryEngine.recovery.sourcePoint', [
```
Can we keep the same directory structure?

fraszczakszymon commented on extensions/wikia/AdEngine/js/provider/gpt/helper.js:
```diff
@@ -55,48 +57,47 @@ define('ext.wikia.adEngine.provider.gpt.helper', [
+		// add adonis marker needed for PF recovery
+		// basing on extra.isPageFairRecoverable param set in factoryWikiaGpt
+		if (isRecoveryEnabled && isBlocking && extra.isPageFairRecoverable) {
+			pageFair.addMarker(element.node);
```
Consider moving it to `factoryWikiaGpt`

fraszczakszymon commented on extensions/wikia/AdEngine/js/provider/yavliTag.js:
```diff
@@ -1,18 +1,18 @@
 /*global define*/
 define('ext.wikia.adEngine.provider.yavliTag', [
```
What do you think about removing whole yavli? Worth to ask Nef/Nandy and make some small clean up :)

fraszczakszymon commented on extensions/wikia/AdEngine/js/slot/revcontentSlots.js:
```diff
@@ -1,10 +1,10 @@
 /*global define*/
 define('ext.wikia.adEngine.slot.revcontentSlots', [
```
What do you think about removing whole revcontent? Worth to ask Nef/Nandy and make some small clean up :)

fraszczakszymon commented on extensions/wikia/AssetsManager/config.php:
```diff
@@ -114,10 +114,10 @@
+		'//extensions/wikia/ARecoveryEngine/js/SourcePoint/sourcePoint.js',
+		'//extensions/wikia/ARecoveryEngine/js/SourcePoint/slotFinder.js',
+		'//extensions/wikia/ARecoveryEngine/js/tweaker.js',
+		'//extensions/wikia/ARecoveryEngine/js/adBlockDetection.js',
```
Keep alphabetical order please

dianafa committed:
Merge branch 'dev' into ADEN-5041

dianafa committed:
ADEN-5041

split adBlockDetection module to detection and adBlockRecovery

dianafa committed:
ADEN-5041

remove error in console that adContext is undefined

dianafa committed:
ADEN-5041

remove yavli

dianafa committed:
ADEN-5041

remove revcontent provider

dianafa committed:
ADEN-5041

fix tests after splitting adBlock recovery & detection

dianafa committed:
ADEN-5041

keep alhpabetical order

dianafa committed:
ADEN-5041

make module name reflect its path

fraszczakszymon commented on extensions/wikia/ARecoveryEngine/js/SourcePoint/slotFinder.js:
```diff
@@ -1,5 +1,5 @@
 /*global define*/
-define('ext.wikia.aRecoveryEngine.recovery.slotFinder', [
+define('ext.wikia.aRecoveryEngine.recovery.sourcePoint.sourcePoint.slotFinder', [
```
Why double `sourcePoint`?

fraszczakszymon commented on extensions/wikia/ARecoveryEngine/js/adBlockDetection.js:
```diff
@@ -0,0 +1,62 @@
+
+	function isBlocking() {
+		return isDetectionEnabled() &&
+			(sourcePoint.isBlocking() || (pageFair && pageFair.isBlocking()));
```
Do we want to rely on pageFair here?

fraszczakszymon commented on extensions/wikia/ARecoveryEngine/js/adBlockDetection.js:
```diff
@@ -0,0 +1,62 @@
+			(sourcePoint.isBlocking() || (pageFair && pageFair.isBlocking()));
+	}
+
+	function isDetectionEnabled() {
```
s/isDetectionEnabled/isEnabled

fraszczakszymon commented on extensions/wikia/ARecoveryEngine/js/adBlockRecovery.js:
```diff
@@ -0,0 +1,95 @@
+			enabled = context.opts.sourcePointRecovery || context.opts.pageFairRecovery;
+
+		log(['isRecoveryEnabled', enabled], log.levels.debug, logGroup);
+		return enabled;
```
`return adBlockDetection.isEnabled() && enabled`

fraszczakszymon commented on extensions/wikia/ARecoveryEngine/templates/ARecoveryEngineApiController_getPageFairBootstrap.mustache:
```diff
@@ -1,13 +1,10 @@
+			window.ads.context &&
+			window.ads.context.opts.pageFairRecovery === true) {
+		{{{code}}}
+	}
```
IMO it does not solve the issue, it hides the race condition. Instead of throwing an error it won't enable pagefair recovery which will be harder to notice. What do you think about switching this line:
https://github.com/Wikia/app/blob/dev/skins/oasis/modules/templates/Oasis_Index.php#L37
(with 38)

fraszczakszymon commented on extensions/wikia/AdEngine/AdEngine2.setup.php:
```diff
@@ -51,7 +50,6 @@
 
 AdEngine2Resource::register('wikia.ext.adengine.pf.detection', 'ResourceLoaderAdEnginePageFairDetectionModule');
 AdEngine2Resource::register('wikia.ext.adengine.sp.detection', 'ResourceLoaderAdEngineSourcePointDetectionModule');
-AdEngine2Resource::register('wikia.ext.adengine.yavli', 'ResourceLoaderAdEngineYavliModule');
```
:gem:

dianafa committed:
ADEN-5041 change order of scipts in oasis_index to avoid racecondition

dianafa committed:
ADEN-5041

do not take into account PF detection, in adBlockRecovery check if adBlockDetection.isEnabled()

nandy-andy commented on extensions/wikia/ARecoveryEngine/js/PageFair/pageFair.js:
```diff
@@ -33,10 +33,10 @@ define('ext.wikia.aRecoveryEngine.recovery.pageFair', [
-
-		slot.setAttribute('adonis-marker', '');
+	function addMarker(slotElement) {
+		if (slotElement) {
```
Is the `if` really needed? We always [create an element](https://github.com/Wikia/app/pull/12808/files#diff-d0456a56fa8978685b777a3c10aaf5abR88) before we call `addMarker()` with it, don't we?

dianafa replied:
yes, we should, but my thinking was not to throw an error in case we'll call addMarker and `doc.getElementById()` which will return null. But this may be overprotective and totally unneeded here. No strong feelings about that from my side.

nandy-andy replied:
Let's remove it then and ask @bulaszenko to do his best to find a way for throwing the error ;) A weak compromise is to add `else` with logging inside. So, once we notice no markers added when all conditions are met, we'll add logging params and see if a lack of element is an issue.

nandy-andy commented on extensions/wikia/ARecoveryEngine/js/PageFair/pageFair.js:
```diff
@@ -1,11 +1,11 @@
-define('ext.wikia.aRecoveryEngine.recovery.pageFair', [
+define('ext.wikia.aRecoveryEngine.pageFair.pageFair', [
```
What are the reasons to change the module name? Oh, probably it is supposed to be mapped to the path, isn't it? [It looks the same with SourcePoint](https://github.com/Wikia/app/pull/12808/files#diff-c55e0a194e1a5eb559d4321c8e034aeeR2). I'm not a big fan but it's not a big issue, I guess. @fraszczakszymon, @v3rth what do you think?

v3rth replied:
IMO pageFair.pageFair looks terrible :D even if it's based on the directory structure, but I don't have a strong argument against than "I don't like it".

Maybe the problem is that last "pageFair" isn't the best and most described name. What do you thing about rename it? If there is no good enough name, maybe it's too big (it has too many responsibilities)

nandy-andy commented on extensions/wikia/ARecoveryEngine/js/SourcePoint/sourcePoint.js:
```diff
@@ -0,0 +1,52 @@
+	 * @returns {boolean}
+	 */
+	function isEnabled() {
+		var enabled = !!context.opts.sourcePointRecovery && !context.opts.pageFairRecovery;
```
Let's make sure we document this logic outside the code: it's possible to overwrite SourcePoint recovery by enabling PageFair recovery :smiley_cat:

dianafa replied:
https://wikia-inc.atlassian.net/wiki/pages/diffpagesbyversion.action?pageId=66551887&selectedPageVersions=33&selectedPageVersions=32

nandy-andy commented on extensions/wikia/ARecoveryEngine/js/SourcePoint/slotFinder.js:
```diff
@@ -1,5 +1,5 @@
 /*global define*/
-define('ext.wikia.aRecoveryEngine.recovery.slotFinder', [
+define('ext.wikia.aRecoveryEngine.sourcePoint.slotFinder', [
```
Shouldn't we keep an old version until cache invalidates?

dianafa replied:
I'd add fallback only for `ext.wikia.aRecoveryEngine.recovery.sourcePoint` due to this change: https://github.com/Wikia/app/pull/12808/files#diff-939a1bcfebcc0e82ae00ac908560e39cR22

nandy-andy replied:
Right, I alarmed "automatically" because I saw the rename. Sorry about that and thanks! :)

nandy-andy commented on extensions/wikia/ARecoveryEngine/js/tweaker.js:
```diff
@@ -1,22 +1,22 @@
 /*global define*/
-define('ext.wikia.aRecoveryEngine.recovery.tweaker', [
+define('ext.wikia.aRecoveryEngine.tweaker', [
```
What about old version cached?

dianafa committed:
ADEN-5041

add adBlockRecovery.old.js

dianafa committed:
ADEN-5041

remove unnecessary if in addMarker

v3rth commented on extensions/wikia/ARecoveryEngine/js/adBlockDetection.js:
```diff
@@ -0,0 +1,57 @@
+/*global define*/
+define('ext.wikia.aRecoveryEngine.adBlockDetection', [
```
isn't it only SP detection? Let's move it to SP directory

dianafa replied:
the point was to create an interface for all detection no matter what's underneath. For now it's SP only, but it can be anything and the point of this module is just to deliver information if user is blocking ads, no matter who is detecting that

v3rth commented on extensions/wikia/ARecoveryEngine/js/adBlockRecovery.old.js:
```diff
@@ -0,0 +1,100 @@
+				try {
+					var xmlHttp = new XMLHttpRequest();
+					xmlHttp.open('GET', customLogEndpoint + type, true);
+					xmlHttp.send();
```
Please, 'pack it' minimum in some function (even whole ```try ... catch```) it will be much cleaner

v3rth commented on extensions/wikia/ARecoveryEngine/js/spec/adBlockDetection.spec.js:
```diff
@@ -0,0 +1,70 @@
+		);
+	}
+
+	it('isBlocking is true when sourcePointDetection is enabled and SP is blocking', function () {
```
Shouldn't "and SP is blocking" be replaced with "SP detects blocking user" or something similar?

v3rth commented on extensions/wikia/AdEngine/js/provider/gpt/googleTag.js:
```diff
@@ -35,40 +35,52 @@ define('ext.wikia.adEngine.provider.gpt.googleTag', [
-		if (gptCanBeLoaded) {
+			node = doc.getElementsByTagName('script')[0];
+
+		if (canGptBeLoaded()) {
```
<3 <3 <3

v3rth commented on extensions/wikia/VideoHandlers/js/handlers/Ooyala.js:
```diff
@@ -165,9 +166,9 @@ define('wikia.videohandler.ooyala', [
-						sourcePoint.addOnBlockingCallback(initRecoveredPlayer);
-						sourcePoint.addOnNotBlockingCallback(initRegularPlayer);
+					if (adBlockDetection.isEnabled()) {
+						adBlockDetection.addOnBlockingCallback(initRecoveredPlayer);
```
IMO it's still only SP not detection in general,  so the name of module isn't adequate

dianafa committed:
ADEN-5041

add function sendCustomLog

dianafa committed:
ADEN-5041

rename pageFair.js and sourcePoint.js to recovery.js

dianafa committed:
ADEN-5041

change comments in tests

dianafa committed:
ADEN-5041

add module to use function from
